Documentation/AdditionalFunctionality/PushNotifications.md

```diff
@@ -0,0 +1,164 @@
+# Push Notifications
+
+The `NotificationCenter` can handle the registration and sending of **push notifications** to Apple Push Notification Service (APNS) and Firebase Cloud Messaging (FCM). APNS can send out push notifications to iOS, macOS and tvOS devices, while FCM handles messaging for iOS, Android and Web Push.
+
+## Configuration
+
+In order to send push notifications, the Apodini web server needs to be authenticated to at least one push notification service. These services are added to a web server by defining them in the `configuration` computed property.
+
+APNS can be configured using either a `.p8` or a `.pem` certificate. FCM is added using the _service-account.json_ file generated by the Firebase Console.
+
+This example adds APNS using a `.pem` certificate and FCM to the web server.
+
+```swift
+var configuration: Configuration {
+    APNSConfiguration(.pem(pemPath: "certificate.pem"), identifier: "de.tum.in.ase.Example", environment: .sandbox)
+    FCMConfiguration("firebase-adminsdk.json")
+}
+```
+
+The `NotificationCenter` can also be used to register devices. This is useful to retrieve saved devices and send out push notifications to them. This feature requires a connected **[database](./Database.md)** to Apodini. Using the special modifier `addNotifications()` with the `DatabaseConfiguration` will create a new database table called `NotificationDevice` of type `Device`.
+
+The following example configures a MongoDB database and enables the `NotificationCenter` to save devices to the _apodini_db_ database.
+
+```swift
+var configuration: Configuration {
+    // ...
+    DatabaseConfiguration(.mongo)
+        .connectionString("mongodb://localhost:27017/apodini_db")
+        .addNotifications()
```

Comment:
I would suggest that the modifier should have additional optional parameters to indicate `Model`s that conform to the `Device` protocol so the developer of Apodini could implement their own `Device`.

Reply:
I've added some documentation for this. In the beginning the modifier will not work with other implementations that conform to `DeviceProtocol` but only with the default implementation of `Device`.